client/my-sites/checkout/composite-checkout/components/contact-details-container.tsx

```diff
@@ -59,15 +57,9 @@ export default function ContactDetailsContainer( {
 	const contactDetails = prepareDomainContactDetails( contactInfo );
 	const contactDetailsErrors = prepareDomainContactDetailsErrors( contactInfo );
 	const { email } = useSelect( ( select ) => select( 'wpcom' ).getContactInfo() );
-	const countries = useSelector( ( state ) => getCountries( state, 'domains' ) );
 
 	const updateDomainContactRelatedData = ( details: DomainContactDetailsData ) => {
 		updateDomainContactFields( details );
-		updateRequiredDomainFields( {
```

Comment:
Removing the postal code requirement seems to retain the postal code requirement when validating for countries that require it, and does not require it for countries that do not.

<img width="172" alt="Screen Shot 2021-09-03 at 5 22 34 PM" src="https://user-images.githubusercontent.com/2036909/132066570-79cb0364-03a1-41c7-af99-7f7a1909899c.png">